presto-hive/src/main/java/io/prestosql/plugin/hive/rcfile/RcFilePageSourceFactory.java

```diff
@@ -134,10 +138,17 @@ else if (deserializerClassName.equals(ColumnarSerDe.class.getName())) {
                 .map(ReaderProjections::getReaderColumns)
                 .orElse(columns);
 
-        FSDataInputStream inputStream;
+        RcFileDataSource dataSource;
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), path, configuration);
-            inputStream = hdfsEnvironment.doAs(session.getUser(), () -> fileSystem.open(path));
+            FSDataInputStream inputStream = hdfsEnvironment.doAs(session.getUser(), () -> fileSystem.open(path));
+            dataSource = new HdfsRcFileDataSource(path.toString(), inputStream, estimatedFileSize, stats);
+            if (estimatedFileSize < BUFFER_SIZE.toBytes()) {
+                byte[] buffer = new byte[toIntExact(estimatedFileSize)];
+                dataSource.readFully(0, buffer, 0, buffer.length);
+                dataSource.close();
```

Comment:
This isn't called if the read throws. I think we can wrap this in try-with-resources